### PR TITLE
Add call & syscall blocks to the program MAST design docs

### PR DESCRIPTION
**Pull request by grjte**
This PR updates the design documentation of the program MAST to include call and syscall blocks

### CODE REVIEW

**Review by bobbinth — CHANGES_REQUESTED** [review]
Looks good! Thank you! A few comments:

1. We should also update the "Program hash computation" section at the end of the file.
2. Minor nit: could we make sizes of the new images a bit smaller?
3. I'm wondering if we should provide additional details about what the VM does when executing `CALL` and `SYSCALL` blocks. For example:
  a. The VM first checks if we are already executing a `syscall` and fails if we do.
  b. The depth of the stack is set at 16.
  c. Upon return, the depth of the stack is checked to be 16, and if it is, the original stack depth is restored. Otherwise, an error happens.

**Comment by grjte** [comment]
Thanks for your review @bobbinth. I've addressed the comments. Please let me know if there's anything else missing!

File: docs/src/design/programs.md [hunk 1]
```diff
@@ -26,6 +26,32 @@ After the body of the loop is executed, the VM checks the top of the stack again
...
 A *loop* block must always have one child, and thus, cannot be a leaf node in the tree.
 
+### Call block
+
+A **call** block is used to describe a function call which is executed in a [user context](../user_docs/assembly/execution_contexts.md). When the VM encounters a *call* block, it creates a new user context, then executes a program which hashes to the target specified by the *call* block in the new context. Thus, in order to execute a *call* block, the VM must be aware of a program with the specified hash. Otherwise, the execution fails. At the end of the *call* block, execution returns to the previous context. 
+
+
+When executing a *call* block, the VM does the following:
+1. Sets the depth of the stack to 16.
+2. Upon return, checks that the depth of the stack is 16. If so, the original stack depth is restored. Otherwise, an error occurs.
```
Comment (bobbinth) [inline docs/src/design/programs.md:36]: Here we also check if we are in a `syscall`. Executing a `call` block inside a `syscall` also results in an error.

**Review by bobbinth — APPROVED** [review]
All looks good! Thank you!

**Pull request merged**